src/attributes/prop.js

```diff
@@ -55,10 +56,19 @@ jQuery.extend( {
 	propHooks: {
 		tabIndex: {
 			get: function( elem ) {
-				return elem.hasAttribute( "tabindex" ) ||
-					rfocusable.test( elem.nodeName ) || elem.href ?
-						elem.tabIndex :
-						-1;
+
+			    // elem.tabIndex doesn't always return the
+			    // correct value when it hasn't been explicitly set
+			    // http://fluidproject.org/blog/2008/01/09/getting-setting-and-removing-tabindex-values-with-javascript/
```

Comment:
I'll switch it to the Web Archive one in a followup. :) We have the same link on compat so we might need to fix both.